src/includes/class-alma-wc-payment-gateway.php

```diff
@@ -821,14 +833,12 @@ public function process_payment( $order_id ) {
 			);
 		}
 
+		// We ignore the nonce verification because process_payment is called after validate_fields.
 		try {
-			$fee_plan_definition = $this->get_fee_plan_definition( $this->checkout_helper->get_chosen_alma_fee_plan() );
+			$fee_plan_definition = $this->get_fee_plan_definition( $_POST['alma_fee_plan'] ); // phpcs:ignore WordPress.Security.NonceVerification
+
 		} catch ( Exception $e ) {
-			$this->logger->log_stack_trace(
-				'Error while creating payment (getting fee plan definition).',
-				$e,
-				array( 'OrderId' => $order_id )
-			);
+			$this->logger->log_stack_trace( 'Error while creating payment: ', $e );
```

Comment:
why did you removed this informations, order_id is one of the context information we want in the error context to help to debug if necessary ...